tests/qam-updinstall/update_install.pm

```diff
@@ -122,117 +140,146 @@ sub run {
-    add_test_repositories;
+        foreach my $b (@patch_l2, @patch_l3) {
+            if (zypper_call("se -t package -x $b", exitcode => [0, 104]) eq '104') {
+                push(@new_binaries, $b);
+            } else {
+                $installable{$b} = 1;
+            }
+        }
 
-    my $patches = get_patch($incident_id, $repos);
+        for my $package (sort keys %installable) {
+
+            # check if we already skipped it
+            next unless defined $installable{$package};
+
+            # Remove binaries conflicting with the ones that are being tested.
+            my $conflict = has_conflict($package);
+            next unless $conflict;
+            if ($installable{$conflict}) {
+                record_info "CONFLICT!", "$package conflicts with $conflict. Skipping $conflict.";
+                delete $installable{$conflict};
+                @patch_l3 = grep !/$conflict/, @patch_l3;
+                @patch_l2 = grep !/$conflict/, @patch_l2;
+            } else {
+                record_info "CONFLICT!", "$package conflicts with $conflict. Removing $conflict.";
+                zypper_call("rm $conflict", exitcode => [0, 104]);
+            }
+        }
 
-    # Check if the patch was correctly configured.
-    # Get info about the patches included in the update.
-    my @patchinfo = split '\n', script_output("zypper -n info -t patch $patches", 200);
-    # Find the lines where the Conflict sections begins.
-    foreach (0 .. $#patchinfo) {
-        print "$_: $patchinfo[$_]\n";
-    }
-    my @conflict_indexes = grep { $patchinfo[$_] =~ /^Conflicts\D*(\d+)/ } 0 .. $#patchinfo;
-    print "conflict_indexes @conflict_indexes\n";
-    # Find the ranges where there are conflict sections.
-    my @ranges = map { $patchinfo[$_] =~ /Conflicts\D*(?<num>\d+)/; ($_ + 1, $_ + $+{num}) } @conflict_indexes;
-    print "ranges @ranges\n";
-    # Make a list of the conflicting binaries.
-    my @conflict_names = uniq pairmap {
-        map { $_ =~ /(^\s+(?<with_ext>\S*)(\.\S* <))|^\s+(?<no_ext>\S*)/; $+{with_ext} // $+{no_ext} } @patchinfo[$a .. $b] } @ranges;
-    print "Conflict names: @conflict_names\n";
-    # Get the l3 released binaries. Only installed binaries can conflict.
-    my @installable_l3 = grep { $bins{$_}->{supportstatus} eq 'l3' } keys %installable;
-    # If not all l3 released binaries are in the conflict binaries, fail.
-    print "\nInstallable L3 @installable_l3\n";
-    for my $package (@installable_l3) {
-        my $hit = first { $package eq $_ } @conflict_names;
-        unless ($hit) {
-            record_info "Error", "$package is l3 but does not exist in the patch. The update may have been misconfigured";
+        # Install released version of installable binaries.
+        # Make sure on SLE 15+ zyppper 1.14+ with '--force-resolution --solver-focus Update' patched binaries are installed
+        my $solver_focus = $zypper_version >= 14 ? '--force-resolution --solver-focus Update ' : '';
```

Comment:
Can [check_version](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/version_utils.pm#L178) be used here? And using a regex to match 1.14 instead of merely 14?

Reply:
I don't know. This works